urlscan/urlchoose.py

```diff
@@ -462,8 +463,13 @@ def browse(*args):
                     self.enter = False
             elif not self.run:
                 webbrowser.open(url)
+            elif self.run and self.pipe:
+                process = Popen(self.run.format(url), shell=True, stdin=PIPE)
+                process.communicate(input=url.encode('utf8'))
```

Comment:
Should this be hardcoded as 'utf8', or should it just be `url.encode()` which I believe will use the default encoding for the user. I think. Encoding/decoding is challenging!

Reply:
Of course, you are right, this should not be hard coded.
But is seems calling it without arguments would not be any better because it defaults to utf8. How about passing the result of `sys.getdefaultencoding()`?

Reply:
Yep... `input=url.encode(sys.getdefaultencoding())`. Added `import sys`.